### ISSUE TITLE
Improved Factory Sorting

### ISSUE THREAD

**Issue by sm0ze** [mekanism/Mekanism#6338]
Improve sorting in the Mekanism Factories by including a "buffer" slot
![Sort](https://user-images.githubusercontent.com/31851788/87644940-23ea5800-c790-11ea-8464-2de09b93725a.png)

With auto-sort enabled:
This would be a new slot inside the machine that is the first priority on item insertion to the machine. It would be separate from the current factory "processing" slots and allow new items to enter the processing slots.

For example:
The basic smelting factory can hold 3 stacks in its processing slots.
If there were 2 stacks worth or less of iron dust all processing slots are being used and it causes backup if you have gold dust waiting for input.
With the "buffer" slot the gold dust can be input to the machine. All processing slots are still iron dust but now there is also gold dust in the machine.
The factory would then collate the iron dust into 2 slots so that the new gold dust can be added to the 3rd processing slot. 
This way there is no backlog of gold dust and all processing slots are still used.

Without auto-sort enabled:
"buffer" slot is greyed out and current factory input logic is used.

TL;DR
Way to input a new type of item to the factory without having to manually turn off sorting and shuffle items around before turning it back on
